tests/Traits/DataHelpers/ProductHelper.php

```diff
@@ -5,23 +5,34 @@
 namespace FINDOLOGIC\FinSearch\Tests\Traits\DataHelpers;
 
 use FINDOLOGIC\FinSearch\Utils\Utils;
-use Psr\Container\ContainerInterface;
 use Shopware\Core\Checkout\Test\Payment\Handler\SyncTestPaymentHandler;
 use Shopware\Core\Content\Product\ProductEntity;
 use Shopware\Core\Defaults;
 use Shopware\Core\Framework\Context;
-use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
 use Shopware\Core\Framework\DataAbstractionLayer\Exception\InconsistentCriteriaIdsException;
 use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
 use Shopware\Core\Framework\Uuid\Uuid;
-use Shopware\Core\System\SalesChannel\Context\SalesChannelContextFactory;
-use Shopware\Core\System\SalesChannel\SalesChannelContext;
-use Shopware\Core\System\SalesChannel\SalesChannelEntity;
 
 trait ProductHelper
 {
-    public function createTestProduct(array $data = [], bool $withVariant = false): ?ProductEntity
+    public function createVisibleTestProduct(array $overrides = []): ?ProductEntity
     {
+        return $this->createTestProduct(array_merge([
+            'visibilities' => [
+                [
+                    'id' => Uuid::randomHex(),
+                    'salesChannelId' => Defaults::SALES_CHANNEL,
+                    'visibility' => 20
+                ]
+            ]
+        ], $overrides));
+    }
+
+    public function createTestProduct(
+        array $data = [],
+        bool $withVariant = false,
+        bool $doRecursive = false
```

Comment:
`$doRecursive` isn't really telling anything. Either rename the variable or add a description in the docblock.